### DISCUSSION TITLE
Off by one bug in sheets.py?

### DISCUSSION THREAD

**Discussion by frosencrantz** [saulpw/visidata#2480] [Q&A] [answered]
Is there an off-by-one bug here?:

https://github.com/saulpw/visidata/blob/8f823f5a3d922756e49ed47086419ab366e8fbc4/visidata/sheets.py#L925-L926

I sometimes get an exception from the scr.addstr(): `error: addwstr() returned ERR`

I tried removing the `=` it seems to work better, though I have not tried it long enough to be sure.

**Comment by midichef** [comment]
Did you see a traceback showing that the exception came from this line? Because the `<=` operator here seems fine to me when testing with default settings.

I have tested that `<= self.windowWidth` works when hitting the case of `==` and not just `<`, by resizing my terminal window with the mouse to put the column separator `|` in the rightmost square of the terminal.

Do you have an unusual setting for `disp_column_sep`? Or maybe an unusual terminal emulator or column width in your data?

There have been other similar errors from other `addstr()` calls trying to draw offscreen, like https://github.com/saulpw/visidata/issues/2171, which is a very low priority bug for now. So I'd like to be sure the error came from line 925 before investigating further.

**Reply by frosencrantz** [reply]
The exception I’m seeing is `_curses.error: addwstr() returned ERR`

The stack trace to get there is sheets.py:798 
 `y += self.drawRow(scr, row, self.topRowIndex+rowidx, y, rowcattr, maxheight=self.windowHeight-y-1, **drawparams)`

From mainloop.py line 28 callNoExceptions return func(*args, **kwargs)


Also, sepchars (and `disp_column_sep`) is an empty string, I'm trying to get a lot of single character values on the screen, though each one is in a separate column.

**Reply by midichef** [reply, marked as answer]
Thanks for the detail. The cause was the empty column separator. I've submitted a fix, #2491.

**Reply by frosencrantz** [reply]
For completeness,  this fixed the issue.
